kobo_selenium_tests/edited_test_cases/enter_data_test_edited.py

```diff
@@ -117,6 +117,17 @@ def test_enter_data_test_edited(self):
         self.assertEqual("Submissions (1)", driver.find_element_by_css_selector(
             ".dashboard__submissions .dashboard__group-label").text)
 
+        # Verify that the QR code is present.
+        driver.find_element_by_css_selector(".dashboard__button-how-to-collect").click()
+        for i in range(60):
+            try:
+                if "How to Collect Data on a Mobile Device" == driver.find_element_by_css_selector("div.vex-content.vex-content-large > #popupmodal > div.vex-header > h3").text: break
```

Comment:
This part is just waiting for the modal to appear, the real test is below on line 129.

> On Oct 20, 2015, at 8:48 PM, dorey <notifications@github.com> wrote:
> 
> In kobo_selenium_tests/edited_test_cases/enter_data_test_edited.py:
> 
> > @@ -117,6 +117,17 @@ def test_enter_data_test_edited(self):
> >          self.assertEqual("Submissions (1)", driver.find_element_by_css_selector(
> >              ".dashboard__submissions .dashboard__group-label").text)
> >  
> > +        # Verify that the QR code is present.
> > +        driver.find_element_by_css_selector(".dashboard__button-how-to-collect").click()
> > +        for i in range(60):
> > +            try:
> > +                if "How to Collect Data on a Mobile Device" == driver.find_element_by_css_selector("div.vex-content.vex-content-large > #popupmodal > div.vex-header > h3").text: break
> In this case, I was referring to testing the qrcode endpoint:
> 
> `/[username]/forms/[formid]/qrcode`
> When this URL returns a 500 error, it suggests that the enketo server is not linked properly or the QR code cant be generated. And the test above would not catch this because the popup would still appear, but I think the qrcode box would just be empty.
> 
> —
> Reply to this email directly or view it on GitHub.
>